common/djangoapps/static_replace/test/test_static_replace.py

```diff
@@ -438,44 +460,60 @@ def create_arbitrary_content(cls, prefix, name, locked=False):
         (u'dev', u'/{c4x}/{prfx}_excluded.html', u'/{c4x}/{prfx}_excluded.html', 1),
         (u'dev', u'/{c4x}/{prfx}_not_excluded.htm', u'/{c4x}/{prfx}_not_excluded.htm', 1),
         # Thumbnails.
-        (u'', u'/{th_key}@{prfx}_unlock-{th_ext}', u'/{th_key}@{prfx}_unlock-{th_ext}', 1),
-        (u'', u'/{th_key}@{prfx}_lock-{th_ext}', u'/{th_key}@{prfx}_lock-{th_ext}', 1),
-        (u'dev', u'/{th_key}@{prfx}_unlock-{th_ext}', u'//dev/{th_key}@{prfx}_unlock-{th_ext}', 1),
-        (u'dev', u'/{th_key}@{prfx}_lock-{th_ext}', u'//dev/{th_key}@{prfx}_lock-{th_ext}', 1),
+        (u'', u'/{base_th_key}@{prfx}_unlock-{th_ext}', u'/{th_key}@{prfx}_unlock-{th_ext}', 1),
+        (u'', u'/{base_th_key}@{prfx}_lock-{th_ext}', u'/{th_key}@{prfx}_lock-{th_ext}', 1),
+        (u'dev', u'/{base_th_key}@{prfx}_unlock-{th_ext}', u'//dev/{th_key}@{prfx}_unlock-{th_ext}', 1),
+        (u'dev', u'/{base_th_key}@{prfx}_lock-{th_ext}', u'//dev/{th_key}@{prfx}_lock-{th_ext}', 1),
     )
     @ddt.unpack
     def test_canonical_asset_path_with_new_style_assets(self, base_url, start, expected, mongo_calls):
         exts = ['.html', '.tm']
         prefix = 'split'
         encoded_base_url = quote_plus('//' + base_url)
         c4x = 'c4x/a/b/asset'
-        asset_key = 'asset-v1:a+b+{}+type@asset+block'.format(prefix)
+        base_asset_key = 'asset-v1:a+b+{}+type@asset+block'.format(prefix)
+        adjusted_asset_key = base_asset_key
         encoded_asset_key = quote_plus('/asset-v1:a+b+{}+type@asset+block@'.format(prefix))
-        th_key = 'asset-v1:a+b+{}+type@thumbnail+block'.format(prefix)
+        base_th_key = 'asset-v1:a+b+{}+type@thumbnail+block'.format(prefix)
+        adjusted_th_key = base_th_key
         th_ext = 'png-16x16.jpg'
 
         start = start.format(
             prfx=prefix,
             c4x=c4x,
-            asset=asset_key,
+            base_asset=base_asset_key,
+            asset=adjusted_asset_key,
             encoded_base_url=encoded_base_url,
             encoded_asset=encoded_asset_key,
-            th_key=th_key,
+            base_th_key=base_th_key,
+            th_key=adjusted_th_key,
             th_ext=th_ext
         )
+
+        # Adjust for content digest.
+        digest = CanonicalContentTest.get_content_digest_for_asset_path(prefix, start)
+        if digest:
+            adjusted_asset_key = 'assets/courseware/[a-f0-9]{{32}}/asset-v1:a+b+{}+type@asset+block'.format(prefix)
+            adjusted_th_key = 'assets/courseware/[a-f0-9]{{32}}/asset-v1:a+b+{}+type@thumbnail+block'.format(prefix)
+            encoded_asset_key = quote_plus('/assets/courseware/MARK/asset-v1:a+b+{}+type@asset+block@'.format(prefix))
+            encoded_asset_key = encoded_asset_key.replace('MARK', '[a-f0-9]{32}')
```

Comment:
Couldn't you just insert the digest in the `encoded_asset_key`, rather than a regex matcher? Then you could continue to use `assertEqual` below, which might simplify things.